plugins/zoom/icon_zoom/util/api.py

```diff
@@ -90,19 +67,67 @@ def get_user_activity_events(
             else:
                 return events
 
+    def _refresh_oauth_token(self):
```

Comment:
This function will either fail entirely without any retries or succeed and does not re-use the _call_api function, which should eliminate any chance of getting into a retry loop.